### PR TITLE
fixes #14096 - The problem with statistics page: Unknown string.

### PR DESCRIPTION
**Pull request by sandersr**
This simple change fixes statistics page on Foreman 1.11.0.RC1

### CODE REVIEW

**Comment by domcleal** [comment]
Hi @sandersr, thanks for the pull request.  As per the bot comment above, would you mind [filing a bug in Redmine](http://projects.theforeman.org/projects/foreman/issues/new) detailing what the exact issue you're seeing is?  If it's a new issue in 1.11 it's especially important for us to know so we can ensure the fix goes out in the right release.

[test] - this'll trigger our existing test suite.  We normally ask for a new test with patches to ensure this doesn't regress later, but I'll need to understand the issue a bit more before I can suggest where/what that should look like.

**Comment by sandersr** [comment]
Hi,

I've read the wiki, created an account, created a bug and updated the commit. Hope it's better now.

**Comment by domcleal** [comment]
Thanks, the stack trace is useful, it suggests that the MiB terminology is coming from facts, i.e. facter/ohai, depending on which you use.  Did you happen to figure out what changed to cause this?

**Comment by sandersr** [comment]
Unfortunately no. I've just migrated from Foreman 1.10 and puppet 3.8 to foreman 1.11.0RC1 and puppet 4.3.2 and I'm trying to make everything working. This was the first problem I encountered.

**Comment by sandersr** [comment]
Looking at facts file:

cat /opt/puppetlabs/server/data/puppetserver/yaml/facts/ovirt.XXXX.yaml 
--- !ruby/object:Puppet::Node::Facts
name: ovirt.XXXX
values:
  architecture: x86_64
  augeas:
    version: 1.4.0
  augeasversion: 1.4.0
  bios_release_date: 04/01/2014
  bios_vendor: SeaBIOS
  bios_version: seabios-1.7.5-11.el7
  blockdevice_sr0_model: QEMU DVD-ROM
  blockdevice_sr0_size: 1073741312
  blockdevice_sr0_vendor: QEMU
  blockdevice_vda_size: 26843545600
  blockdevice_vda_vendor: '6900'
  blockdevices: sr0,vda
  chassistype: Other
  disks:
    sr0:
      model: QEMU DVD-ROM
      size: 1.00 GiB
      size_bytes: 1073741312
      vendor: QEMU
    vda:
      size: 25.00 GiB
      size_bytes: 26843545600
      vendor: '6900'
  dmi:
    bios:
      release_date: 04/01/2014
      vendor: SeaBIOS
      version: seabios-1.7.5-11.el7
    chassis:
      type: Other
    manufacturer: oVirt
    product:
      name: oVirt Node
      serial_number: 4C4C4544-0048-3810-8046-XXXXXXXXXXXX
      uuid: 3861F747-8EBA-6344-A778-89D9XXXXXXXX
  domain: XXX
  facterversion: 3.1.4
  filesystems: ext4,iso9660
  fqdn: ovirt.teamwpc.local
  gid: root
  hardwareisa: x86_64
  hardwaremodel: x86_64
  hostname: ovirt
  id: root
  identity:
    gid: 0
    group: root
    uid: 0
    user: root
  interfaces: eth0,lo
  ipaddress: 192.168.XX.XX
  ipaddress6: fe80::216:3eff:fe75:61b6
  ipaddress6_eth0: fe80::216:3eff:fe75:61b6
  ipaddress6_lo: '::1'
  ipaddress_eth0: 192.168.XX.XX
  ipaddress_lo: 127.0.0.1
  is_virtual: true
  kernel: Linux
  kernelmajversion: '2.6'
  kernelrelease: 2.6.32-573.18.1.el6.x86_64
  kernelversion: 2.6.32
  load_averages:
    15m: 0.08
    1m: 0.0
    5m: 0.08
  macaddress: 00:16:3e:XX:XX:XX
  macaddress_eth0: 00:16:3e:XX:XX:XX
  manufacturer: oVirt
  memory:
    swap:
      available: 1.96 GiB
      available_bytes: 2101252096
      capacity: 21.72%
      total: 2.50 GiB
      total_bytes: 2684350464
      used: 556.09 MiB
      used_bytes: 583098368
    system:
      available: 889.30 MiB
      available_bytes: 932495360
      capacity: 76.80%
      total: 3.74 GiB
      total_bytes: 4018692096
      used: 2.87 GiB
      used_bytes: 3086196736
  memoryfree: 889.30 MiB
  memoryfree_mb: 889.296875
  memorysize: 3.74 GiB
  memorysize_mb: 3832.5234375
  mountpoints:

**Comment by domcleal** [comment]
Right, that makes sense, a newer version of Facter.

Could you add a test for the MiB parsing to test/lib/core_extensions_test.rb please?  You can copy the existing to_gb test for `1024 MB`, just change the test name and the tested value to use `MiB` instead.

File: lib/core_extensions.rb [hunk 1]
```diff
@@ -135,13 +135,13 @@ def to_translation
...
-      value, _, unit=self.match(/(\d+(\.\d+)?) ?(([KMGT]B?|B))$/i)[1..3]
+      value, _, unit=self.match(/(\d+(\.\d+)?) ?(([KMGT]i?B?|B))$/i)[1..3]
       case unit.to_sym
       when nil, :B, :byte          then (value.to_f / Foreman::SIZE[:giga])
-      when :TB, :T, :terabyte      then (value.to_f * Foreman::SIZE[:kilo])
-      when :GB, :G, :gigabyte      then value.to_f
-      when :MB, :M, :megabyte      then (value.to_f / Foreman::SIZE[:kilo])
-      when :KB, :K, :kilobyte, :kB then (value.to_f / Foreman::SIZE[:mega])
+      when :TB, :TiB, :T, :terabyte      then (value.to_f * Foreman::SIZE[:kilo])
+      when :GB, :Gib, :G, :gigabyte      then value.to_f
```
Comment (domcleal) [inline lib/core_extensions.rb]: Should this be `GiB` (capital 'B')?

**Comment by domcleal** [comment]
Added an inline comment too - when updating, please just squash your commits together, so make the change, `git add` and `git commit --amend` + push.

File: lib/core_extensions.rb (hunk 1 above)
Comment (sandersr) [inline lib/core_extensions.rb]: Yes, thanks for spotting it!

I'll add a test in a sec and update the commit.

**Comment by sandersr** [comment]
Not having much luck with git...

**Comment by domcleal** [comment]
Probably the way to fix it is to:

```
git rebase -i upstream/develop   # assuming theforeman is upstream, project etc.
# change the "pick" to "fixup" at the start of the line for the second commit
edit lib/core_extensions.rb and fix the conflict around `>>>>>` - just keep the `GiB` line
git rebase --continue
```

and you should end up with one commit, just `git push -f origin` that then.

If you run into trouble, don't worry too much, I can fix it when merged.

**Comment by sandersr** [comment]
Your help is greatly appreciated. Followed your instructions and it looks good!

File: test/lib/core_extensions_test.rb [hunk 2]
```diff
@@ -8,6 +8,13 @@ class CoreExtensionsTest <  ActiveSupport::TestCase
       assert_equal(1.0, to_gb_value, "Converted value should be 1.0")
       assert(to_gb_value.is_a?(Float), "Converted value shoud be a float")
     end
+
```
Comment (dLobatog) [inline test/lib/core_extensions_test.rb:11]: There's some trailing whitespace here rubocop is not liking

Comment (domcleal) [inline test/lib/core_extensions_test.rb:11]: Ah yes, I fixed that on merge.

**Comment by domcleal** [comment]
Merged as b24feca, thanks @sandersr!

**Pull request closed without merging**